faker/providers/ssn/es_MX/__init__.py

```diff
@@ -0,0 +1,254 @@
+    "KAGA": "KAGX",
+    "KAGO": "KAGX",
+    "KOGE": "KOGX",
+    "KOJO": "KOJX",
+    "KAKA": "KAKX",
+    "KULO": "KULX",
+    "MAME": "MAMX",
+    "MAMO": "MAMX",
+    "MEAR": "MEAX",
+    "MEAS": "MEAX",
+    "MEON": "MEOX",
+    "MION": "MIOX",
+    "MOCO": "MOCX",
+    "MULA": "MULX",
+    "PEDA": "PEDX",
+    "PEDO": "PEDX",
+    "PENE": "PENX",
+    "PUTA": "PUTX",
+    "PUTO": "PUTX",
+    "QULO": "QULX",
+    "RATA": "RATX",
+    "RUIN": "RUIN",
+}
+
+CURP_CHARACTERS = "0123456789ABCDEFGHIJKLMNÑOPQRSTUVWXYZ"
+
+
+def _reduce_digits(number):
+    """
+    Sum of digits of a number until sum becomes single digit.
+
+    Example:
+        658 => 6 + 5 + 8 = 19 => 1 + 9 = 10 => 1
+    """
+    if number == 0:
+        return 0
+    if number % 9 == 0:
+        return 9
+
+    return number % 9
+
+
+def ssn_checksum(digits):
+    """
+    Calculate the checksum for the mexican SSN (IMSS).
+    """
+    return -sum(
+        _reduce_digits(n * (i % 2 + 1))
+        for i, n in enumerate(digits)
+    ) % 10
+
+
+def curp_checksum(characters):
+    """
+    Calculate the checksum for the mexican CURP.
+    """
+    start = 18
+    return -sum(
+        (start - i) * CURP_CHARACTERS.index(n)
+        for i, n in enumerate(characters)
+    ) % 10
+
+
+class Provider(BaseProvider):
+    """
+    A Faker provider for the Mexican SSN, RFC and CURP
+    """
+    ssn_formats = ("###########",)
+
+    def ssn(self):
+        """
+        Mexican Social Security Number, as given by IMSS.
+
+        :return: a random Mexican SSN
+        """
+        office = self.random_int(min=1, max=99)
+        birth_year = self.random_int(min=0, max=99)
+        start_year = self.random_int(min=0, max=99)
+        serial = self.random_int(min=1, max=9999)
+
+        num = "{0:02d}{1:02d}{2:02d}{3:04d}".format(
+            office,
+            start_year,
+            birth_year,
+            serial,
+        )
+
+        check = ssn_checksum(map(int, num))
+        num += str(check)
+
+        return num
+
+    def curp(self):
+        """
+        See https://es.wikipedia.org/wiki/Clave_%C3%9Anica_de_Registro_de_Poblaci%C3%B3n.
+
+        :return: a random Mexican CURP (Unique Population Registry Code)
+        """
+        birthday = self.generator.date_of_birth()
+
+        first_surname = random.choice(ALPHABET) + random.choice(VOWELS)
+        second_surname = random.choice(ALPHABET)
+        given_name = random.choice(ALPHABET)
+        name_initials = first_surname + second_surname + given_name
+
+        birth_date = birthday.strftime("%y%m%d")
+        gender = random.choice("HM")
+        state = random.choice(STATES_RENAPO)
+        first_surname_inside = random.choice(CONSONANTS)
+        second_surname_inside = random.choice(CONSONANTS)
+        given_name_inside = random.choice(ALPHABET)
+
+        # This character is assigned to avoid duplicity
+        # It's normally '0' for those born < 2000
+        # and 'A' for those botn >= 2000
```

Comment:
Typo here I think, this should be `born` I think

Reply:
Thanks, fixed!